### PR TITLE
Added memproof for devnet

### PR DESCRIPTION
**Pull request by zeeshan595**
@zo-el Could you please have a look. 
* I am not sure if the correct environment variables are being passed to this process
* Let me know if you want me to restructure the code

This PR enables membrane proof after the randomly generated key is created for devNet. This is because originally membrane proof is done before this key is generated and because of this it can never succeed.

### CODE REVIEW

**Comment by alastairong** [comment]
Looks like you have to update the cargo file with `failure` crate?

File: src/websocket.rs
Comment (alastairong) [inline src/websocket.rs:16]: No longer necessary?

File: src/websocket.rs
Comment (alastairong) [inline src/websocket.rs:2]: Need to import crate

File: src/websocket.rs
Comment (alastairong) [inline src/websocket.rs:140]: I don't understand why we're doing any of this. Configure-holochain shouldn't need to talk to the memproof server itself?

**Review by alastairong — COMMENTED** [review]
Hey let's chat about this after standup.

File: src/websocket.rs
Comment (zeeshan595) [reply src/websocket.rs:139]: @alastairong 
From what Joel told me. Membrane proof can't get the generated key on dev net because it is created after it requests membrane proof. So Joel said the simplest solution is probably to request the membrane proof (only for devNet) after the key is geenrated which is in `configure holochain`

Comment (alastairong) [reply src/websocket.rs:139]: Oh jeez now I get it. Wow this is really brute forcing it huh?

So in devNet require requesting a second memproof with the real agentId in configure holochain?

**Comment by alastairong** [comment]
@zo-el I had no idea this was what you were requesting. It seems very heavy handed and I wonder if we should just enforce "static" agent key in devNet like in production instead...

I request we wait until Monday to merge so that we can get PJ's input as well.

File: src/websocket.rs
Comment (zo-el) [reply src/websocket.rs:139]: It is not brute-forcing it because its should only generate it when the random key was created

File: src/websocket.rs
Comment (zo-el) [inline src/websocket.rs:188]: you don't need to generate it here, this is where is just reading from a local copy. in the line below you have already created the mem-proof when the key was generated

File: src/websocket.rs
Comment (zo-el) [inline src/websocket.rs:206]: you need to pass the agent_key that is generated here to be signed and used in the mem-proof

**Review by zo-el — CHANGES_REQUESTED** [review]
Not building. 

Also, @alastairong we cannot have static agents for `devNet` because the purpose of creating random agents is to be easy for devs to delete their source chains and not cause an issue in the network.

**Comment by zeeshan595** [comment]
@zo-el I made all of the requested changes and it should be working now. I am not sure the best way to test this PR. I will leave that to you guys until i get further info on testing it :) 

CC: @alastairong

File: src/websocket.rs
Comment (zo-el) [inline src/websocket.rs:195]: You do not need this.

**Review by zo-el — CHANGES_REQUESTED** [review]
You still have to pass the random key that was generated and pass that to the mem-proof-server.

Also, can we keep this clean and have it in a different mod

**Review by zo-el — APPROVED** [review]
Looks good. Nice job :+1:

**Review by peeech — APPROVED** [review]
So my only comment that I can give here is that the code looks pretty. I do not really have any requirements to check against, there are no tests proving any thesis, nor I can even run a binary and assess an outcome (as this binary is a part of a bigger ecosystem). Code compiles, approved

**Pull request merged**
